### ISSUE TITLE
Phase 46.4: RepresentationEvaluator — Representation Quality Assessment

### ISSUE THREAD

**Issue by web3guru888** [web3guru888/asi-build#908]
## Phase 46.4: RepresentationEvaluator

**Parent**: Phase 46 — Self-Supervised Learning & Contrastive Representations (Discussion #904)

### Overview
Implements comprehensive evaluation protocols for assessing the quality of self-supervised learned representations. Provides standardized benchmarks including linear probing, few-shot evaluation, and representation geometry analysis to compare SSL methods objectively.

#### Linear Probing
- Freeze encoder, train linear classifier on top
- SGD with learning rate sweep (0.01–30.0)
- Standard evaluation on ImageNet, CIFAR-10/100, STL-10
- Layer-wise probing (evaluate intermediate representations)
- Training data efficiency curves (1%, 10%, 100% labels)

#### k-Nearest Neighbors Evaluation
- Non-parametric feature quality assessment
- Weighted k-NN classification (k=10, 20, 200)
- Temperature-scaled distance weighting
- No training required — direct feature space evaluation
- Memory-efficient approximate k-NN for large datasets

#### Few-Shot Classification
- Episodic evaluation with N-way K-shot protocol
- 5-way 1-shot and 5-way 5-shot standard benchmarks
- Prototypical network evaluation (class mean matching)
- Meta-dataset evaluation across multiple domains
- Confidence intervals over multiple episodes (600+)

#### Centered Kernel Alignment (CKA)
- Compare representations between layers/models
- Linear CKA: efficient Frobenius norm computation
- RBF kernel CKA: captures nonlinear similarity
- Mini-batch CKA for memory-efficient computation
- Representation similarity matrices and heatmaps

#### Mutual Information Estimation
- MINE (Mutual Information Neural Estimation)
- InfoNCE-based lower bound estimation
- Representation-label mutual information
- Inter-layer information flow analysis
- Compression vs. prediction information trade-off

#### Uniformity and Alignment Metrics
- Alignment: expected distance between positive pairs
- Uniformity: distribution uniformity on hypersphere
- Wang & Isola metrics with temperature parameter
- Dimensional collapse detection (effective rank)
- Feature covariance spectrum analysis

### Interface
```python
class RepresentationEvaluator:
    def evaluate(
        self,
        encoder: nn.Module,
        benchmark: EvalBenchmark,
        protocol: str  # "linear_probe" | "knn" | "few_shot" | "cka" | "full"
    ) -> QualityReport:
        """Evaluate representation quality using specified protocol."""

    def linear_probe(
        self,
        encoder: nn.Module,
        train_data: LabeledDataset,
        test_data: LabeledDataset,
        num_classes: int
    ) -> ProbeResult:
        """Train and evaluate linear classifier on frozen features."""

    def compute_cka(
        self,
        representations_a: Tensor,
        representations_b: Tensor,
        kernel: str = "linear"
    ) -> float:
        """Compute CKA similarity between two representation sets."""

    def measure_geometry(
        self,
        representations: Tensor
    ) -> GeometryMetrics:
        """Measure uniformity, alignment, effective rank, isotropy."""
```

### Acceptance Criteria
- [ ] Linear probing reproduces published baselines (±1% accuracy)
- [ ] k-NN evaluation correlates with linear probe rankings
- [ ] Few-shot evaluation produces tight confidence intervals
- [ ] CKA correctly identifies similar/dissimilar representations
- [ ] Uniformity metric detects dimensional collapse
- [ ] Evaluation protocols are reproducible across runs (fixed seeds)
- [ ] Full evaluation suite runs within 30 minutes on single GPU
- [ ] Unit tests with >90% coverage

### References
- Kornblith et al. (2019) CKA — Similarity of Neural Network Representations Revisited
- Wang & Isola (2020) Understanding Contrastive Representation Learning through Alignment and Uniformity
- Belghazi et al. (2018) MINE — Mutual Information Neural Estimation
- Triantafillou et al. (2020) Meta-Dataset — Benchmark for Few-Shot Classification
